lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,180 @@
+import { logger } from '../../../logger';
+import { ExternalHostError } from '../../../types/errors/external-host-error';
+import { cache } from '../../../util/cache/package/decorator';
+import { HttpError } from '../../../util/http';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { PackageType, datasource, defaultRegistryUrl } from './common';
+
+export class HexpmBobDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+
+  override readonly caching = true;
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? defaultRegistryUrl}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const packageType = this.getPackageType(packageName);
+
+    if (!packageType) {
+      return null;
+    }
+
+    logger.trace(
+      { registryUrl: registryUrl ?? defaultRegistryUrl, packageName },
+      `fetching hex.pm bob ${packageName} release`
+    );
+
+    const url = `${
+      registryUrl ?? defaultRegistryUrl
+    }/builds/${packageName}/builds.txt`;
+
+    const result: ReleaseResult = {
+      releases: [],
+      ...this.getPackageDetails(packageType),
+    };
+    try {
+      const { body } = await this.http.get(url);
+      result.releases = body
+        .split('\n')
+        .map((line) => line.trim())
+        .filter((line) => line !== '')
+        .map((line) => {
+          const [version, gitRef, buildDate, newDigest] = line.split(' ');
+
+          return {
+            changelogUrl: this.changelogUrl(version, packageType),
+            downloadUrl: `${
+              registryUrl ?? defaultRegistryUrl
+            }builds/${packageName}/${version}.tar.gz`,
+            gitRef,
+            isStable: this.isStable(version, packageType),
+            releaseTimestamp: new Date(buildDate).getTime(),
+            version: this.cleanVersion(version, packageType),
+            newDigest,
+            constraints: this.constraints(version, packageType),
+            sourceUrl: this.sourceUrl(gitRef, packageType),
+          };
+        });
+    } catch (err) {
+      if (err instanceof HttpError && err.response?.statusCode !== 404) {
+        throw new ExternalHostError(err);
+      }
+      this.handleGenericErrors(err);
+    }
+
+    return result.releases.length > 0 ? result : null;
+  }
+
+  private getPackageType(packageName: string): PackageType | null {
+    if (packageName === 'elixir') {
+      return 'elixir';
+    }
+    if (packageName.startsWith('otp/')) {
```

Comment:
Those two are the solutions I see:

**a: Always use `ubuntu-20.04`**

The supported `packageName` is `erlang`. The operating system, may not be provided because it is irrelevant. (All operating systems build the same versions. It would only be relevant if the builds/checksums would be used.)

The actual name that is looked up, could be stored as a const in `lib/modules/datasource/hexpm-bob/common.ts`.

**b: Validate the name better**

Unfortunately, there's no manifest file published that lists all the possible operating systems. The ones I know of are:

* `ubuntu-14.04`
* `ubuntu-16.04`
* `ubuntu-18.04`
* `ubuntu-20.04`
* `ubuntu-22.04`
* `alpine-3.10`

Therefore I could do something along the lines of this:

```js
if (packageName.match(/^otp\/[\w]+\-\d+-\d+$/)) {
```

@viceice Should I go with option a or b or do you have a better idea?

Reply:
Went with option b.